xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -624,18 +632,26 @@ private void handleLdsResponseForConfigUpdate(DiscoveryResponse ldsResponse) {
     adsStream.sendAckRequest(ADS_TYPE_URL_LDS, ImmutableList.of(ldsResourceName),
         ldsResponse.getVersionInfo());
 
-    if (clusterName != null || rdsRouteConfigName != null) {
+    if (routes != null || rdsRouteConfigName != null) {
```

Comment:
`routes` would never be `null`. Ditto below.

Reply:
`routes` initially is null. It's assigned conditionally. Why would route never be null?

Reply:
You are right. It's conditionally assigned.